tests/pytest_tests/unit_tests/test_artifacts/test_storage.py

```diff
@@ -36,6 +39,61 @@ def test_opener_rejects_append_mode(artifact_file_cache):
         f.write("example")
 
 
+def test_opener_works_across_filesystem_boundaries(
+    tmp_path, artifact_file_cache, fs: FakeFilesystem
+):
+    # This is not ideal, as we'd much rather test e.g. `Artifact.download()` directly or E2E.
+    #
+    # However, we're using `pyfakefs` to mock mounted/partitioned filesystems, and it doesn't play well with
+    # some of the internals of ArtifactFileCache without additional, potentially brittle/complicated patches
+    # to unrelated operations (e.g. around `subprocess.call()`).  This will have to do for the moment.
```

Comment:
nit: limit line length to 80 chars; these comments wrap around on my screen when reviewing

Reply:
np, can change (just kept my IDE settings at 120 chars since the linter seemed OK with it 😅)